Sources/Build/BuildPlan.swift

```diff
@@ -229,6 +240,17 @@ public final class ClangTargetBuildDescription {
         }
         args += ["-fblocks"]
 
+        // Enable index store, if appropriate.
+        //
+        // Currently, we only enable index store for Apple's clang because it is
+        // not widely available in OSS clang.
+        if buildParameters.shouldEnableIndexStore,
+           buildParameters.configuration == .debug,
+           buildParameters.triple.isDarwin(),
+           (try? buildParameters.toolchain._isClangCompilerVendorApple()) == true {
+            args += ["-index-store-path", buildParameters.indexStore.asString]
```

Comment:
I assume there's no way currently to 'query' the compiler if it supports a compiler option ? (e.g. similar to what CMake can do).

Reply:
Yeah, I don't think there is a good way to detect this right now.